integrations/MULTI_GATEWAY_CLIENT.md

```diff
@@ -0,0 +1,175 @@
+
+### Finding new gateways
+
+The `gateways` file is parsed to know the initial - bootstrap - gateways. Each line in this file is a single gateway. This list of gateways should be stored internally in this `multi gateway client` implementation.
+
+Your internal gateway list might now look like:
+
+```
+http://localhost:8080
+https://ipfs.io
+```
+
+From this point on the client should iterate over those gateways and request each of them to give a list of [gateways that it knows](#Gateway-returns-list-of-gateways-it-knows). Based on the return, this should result in a vastly bigger list of potentially usable gateways:
+
+```
+http://localhost:8080
+N+1
+N+2
+...
+https://ipfs.io
+N+1
+N+2
+...
+```
+
+Important to note here is that this request for more gateways will only be done on the initial list of provided gateways! It's non-recursive and therefore won't be executed on newly found gateways.
+
+### Validating gateways to be potentially used
+
+Each gateway will be tasked to execute a query. The results of this query and the time it took determine if a gateway can potentially be used. A hard requirement for a gateway is to support [trustless](https://docs.ipfs.tech/reference/http/gateway/#trusted-vs-trustless) data retrieval!
+
+```mermaid
+  graph TD;
+      A[Supports trustless?] --> B{Response};
+      B --> |200 Ok - Response within 200ms| C[CAR request];
+      B --> |200 Ok - Response 200+ms| D;
+      B --> |Anything else| D[Disgard gateway];
+      C --> E{Validate CAR};
+      E --> |Invalid| D;
+      E --> G[Valid];
+      G --> H[Store gateway];
+```
+
+The `200ms` threshold here is arbitrarily picked. From a decentralized point of view, 200ms allows you to go roughly halfway across the globe assuming your internet connection is stable. From a data retrieval point of view 200ms can be slow but can be just fine too. For example, if a site loads with 1 connection at a time with each connection having a 200ms latency then you will experience that site to "load slow". But if you load the same site with multiple concurrent connections where "some" might hit the 200ms threshold then you won't see much difference.
+
+Once a gateway, in the above flow, reaches the `Store gateway` point then it should internally in this `multi gateway client` implementation be stored in some array.
+
+### Keeping the usable gateway list fresh in the background
+
+Getting this list of gateways and maintaining if they should be used can take quite some time. The adviced approach here is to run each request in an async matter where the async flow follows the same flow as the above flowchart.
+
+### Configuration options
+
+`concurrent requests` Defaults to 10. There must be a way to specify how many concurrent requests the `multi gateway client` does per IPFS request.
+
+`max simultaneous cids` Defaults to 5. There must be a way to define how many simultaneous IPFS requests the `multi gateway client` can handle at any given time.
```

Comment:
Is there a reason "5" was chosen here? It may make sense to set to 6 also for the reasons [above](https://github.com/ipfs/specs/pull/359/files#r1055893571)